cou/steps/plan.py

```diff
@@ -304,10 +304,34 @@ async def generate_plan(analysis_result: Analysis, args: CLIargs) -> UpgradePlan
     :rtype: UpgradePlan
     """
     pre_plan_sanity_checks(args, analysis_result)
-    hypervisors = await filter_hypervisors_machines(args, analysis_result)
-    logger.info("Hypervisors selected: %s", hypervisors)
     target = determine_upgrade_target(analysis_result)
 
+    plan = generate_common_plan(analysis_result, args, target)
+
+    control_plane_plan = await generate_control_plane_plan(analysis_result, args, target)
```

Comment:
Why not simply do if is data-plane add it as sub-step? Name of function is generate and not get, so it should generate and not return empty step or None.

Reply:
Yes, I agree. If the name of function is `get_*` ok, it can return empty object or None, but not `generate_*`.

Reply:
```python
    plan = generate_common_plan(analysis_result, args, target)

    if args.is_control_plane_command:
        plan.add_step(await generate_control_plane_plan(analysis_result, args, target))

    if args.is_data_plane_command:
        plan.add_step(await generate_data_plane_plan(analysis_result, args))

```
It is cleaner and much obvious what we are trying to do, instead of having to generate returning empty step and then have to loop with  `control_plane_plan + data_plane_plan` and `if group`.